io/shared/src/main/scala/scalaz/effect/IO.scala

```diff
@@ -272,13 +249,13 @@ sealed abstract class IO[E, A] { self =>
    * Runs the specified cleanup action if this action errors, providing the
    * error to the cleanup action. The cleanup action will not be interrupted.
    */
-  final def onError(cleanup: Throwable \/ E => IO[Void, Unit]): IO[E, A] =
+  final def onError(cleanupT: Throwable => IO[Nothing, Unit])(cleanupE: E => IO[Nothing, Unit]): IO[E, A] =
```

Comment:
@jdegoes Now that we have decided to go with `Either` in the API, does this change make sense?